app/src/js/controllers/ImageController.js

```diff
@@ -47,6 +48,44 @@ angular.module('media_manager')
       CourseCache.current_image = CourseCache.images[ic.index];
       resetBreadcrumb();
     }
-  }
+  };
+
+  ic.newLabel = '';
+  ic.newValue = '';
+  ic.saveMetadata = function(label, value){
+    if(label !== undefined && value !== undefined){
+      ic.image.metadata.forEach(function(item, index, arr){
+        if(item.label == label){
+          arr[index].label = label;
+          arr[index].value = value;
+          Image.update({}, ic.image, function success(data){
+
+          }, function failure(errorResponse){
+            $log.debug("error updating image:", errorResponse);
```

Comment:
If the metadata update fails, shouldn't we update the UI accordingly? Otherwise, it will be a silent failure and the user will wonder why their metadata wasn't saved. Since this is called in the context of the **onbeforesave**, which is a validation hook, could we take advantage of that somehow?